src/reducers/root.js

```diff
@@ -0,0 +1,61 @@
+import {
+    REFRESH,
+    CHANGE_GAME_BEGIN,
+    MAKE_SHOOT,
+    GENERATE_FIELD
+} from "../actions/type";
+import Field from "../components/SeaBattle/src/Field";
+
+
+export default function gameState(state, action) {
+    switch (action.type) {
+
+        case REFRESH:
+
+            return {...state, my_field: new Field().createField(),
+                comp_field: new Field().createField(), gameBegin: false};
+
+        case CHANGE_GAME_BEGIN:
+            return {...state, gameBegin: true };
+
+        case MAKE_SHOOT:
+
+            if (!state.gameBegin)
+                return state;
+
+            const { my_field, comp_field } = state;
+
+            let shot = comp_field.shoot(action.row, action.col);
```

Comment:
you are mutating state here, so this could cause problems, cause you need to avoid doing it with redux. The more appropriate way to do it is to return the copy of field but after shooting.